src/components/Search/index.tsx

```diff
@@ -144,15 +145,27 @@ function Search({query, policyIDs, sortBy, sortOrder, isMobileSelectionModeActiv
             SearchActions.createTransactionThread(hash, item.transactionID, reportID, item.moneyRequestReportActionID);
         }
 
-        Navigation.navigate(ROUTES.SEARCH_REPORT.getRoute(query, reportID));
+        Navigation.navigate(ROUTES.SEARCH_REPORT.getRoute(reportID));
     };
 
     const fetchMoreResults = () => {
         if (!searchResults?.search?.hasMoreResults || shouldShowLoadingState || shouldShowLoadingMoreItems) {
             return;
         }
-        const currentOffset = searchResults?.search?.offset ?? 0;
-        SearchActions.search({hash, query, offset: currentOffset + CONST.SEARCH.RESULTS_PAGE_SIZE, sortBy, sortOrder});
+
+        const currentSearchParams = SearchUtils.getCurrentSearchParams();
+        const currentQueryString = SearchUtils.getQueryStringFromParams(currentSearchParams);
+        const isCustomQuery = SearchUtils.isCustomQueryFromParams(currentSearchParams);
+        const currentQueryJSON = SearchUtils.buildSearchQueryJSON(currentQueryString);
```

Comment:
The more I think about `cq` and `q`, the more I think this is not a good solution. We need additional code to work with queries saved in params. Maybe we should have one `q` parameter and another parameter telling us if this query is canne or custom? This should be easier to manage. What do you think guys? @luacmartins @Kicu @WojtekBoman

Reply:
I'm not sure that I understand the issue. What do you mean by ` We need additional code to work with queries saved in params.`?

Reply:
Ah do you mean the methods above, e.g. `getCurrentSearchParams`, `getQueryStringFromParams` and `getQueryStringFromParams`?

Reply:
I made a similar comment, but I wonder if we should just create a separate page `SearchResultsPage` and pass a `isCustomQuery=true` prop to `Search/index.tsx` so we can avoid all this logic and the `QUERY_KIND` stuff we have above.